src/Microsoft.AspNetCore.Authorization/AuthorizationOptions.cs

```diff
@@ -11,7 +11,10 @@ namespace Microsoft.AspNetCore.Authorization
     /// </summary>
     public class AuthorizationOptions
     {
-        private IDictionary<string, AuthorizationPolicy> PolicyMap { get; } = new Dictionary<string, AuthorizationPolicy>(StringComparer.OrdinalIgnoreCase);
```

Comment:
This makes everything mutable and bypasses the argument validation on AddPolicy. What if you added GetPolicies that returned an IEnumerable instead?